vdr/doc/resolvers.go

```diff
@@ -31,17 +32,53 @@ import (
 	"github.com/nuts-foundation/nuts-node/vdr/types"
 )
 
+// ErrNestedDocumentsTooDeep is returned when a DID Document contains a multiple services with the same type
+var ErrNestedDocumentsTooDeep = errors.New("DID Document controller structure has too many indirections")
+
+const maxControllerDepth = 5
```

Comment:
I couldn't find anything in the spec about this. Maybe we can open an issue on https://github.com/w3c/did-core to clarify how we should handle this? I mean `5` is probably fine but it's feels quite arbitrary

Reply:
Yes, 5 is arbitrary chosen based on the expected hierarchy of DID Documents. 3 is to be expected: offlineRoot->activeSaasKeys->customers. 2 has been added as a safe margin. This could also easily be changed to 10 or 100 since that would protect from recursion.

I don't think it's something that should go in the w3c specs since adding a hierarchy using controllers is something we specified. I did sent an email to the public-did-wg mailing list.